File: LeetCode/169. Majority Element.cpp
int majorityElement(vector<int>& nums) {
    map<int, int> mp;
    for (auto it : nums) {
        mp[it++];
    }
    for (auto it = mp.begin(); it != mp.end(); ++it) {
        it -> second = count(nums.begin(), nums.end(), it -> first);
        if (it -> second > nums.size() / 2)
            return it -> first;
    }
    return 0;
}

File: LeetCode/169. Majority Element.go
func majorityElement(nums []int ) int {
    count := 0
    res := 0

    for _, el := range nums {
        if count == 0 {
            res = el
        }
        if el == res {
            count++
        } else {
            count--
        }
    }

    return res
}
